py/selenium/webdriver/common/bidi/browser.py

```diff
@@ -123,16 +123,51 @@ def from_dict(cls, data: dict) -> "ClientWindowInfo":
         Returns:
         -------
             ClientWindowInfo: A new instance of ClientWindowInfo.
+
+        Raises:
+        ------
+            ValueError: If required fields are missing or have invalid types.
         """
-        return cls(
-            client_window=data.get("clientWindow"),
-            state=data.get("state"),
-            width=data.get("width"),
-            height=data.get("height"),
-            x=data.get("x"),
-            y=data.get("y"),
-            active=data.get("active"),
-        )
+        try:
+            client_window = data.get("clientWindow")
+            if not isinstance(client_window, str):
+                raise ValueError("clientWindow must be a string")
+
+            state = data.get("state")
+            if not isinstance(state, str):
+                raise ValueError("state must be a string")
+
+            width = data.get("width")
+            if not isinstance(width, int):
+                raise ValueError("width must be an integer")
```

Comment:
```suggestion
			width = data["width"]
            if not isinstance(width, int) or width < 0:
                raise ValueError(f"width must be a non-negative integer, got {width}")
```
Same here with `.get()`, the spec (https://w3c.github.io/webdriver-bidi/#type-browser-ClientWindowInfo) mentions height and width should be uint (non-negative int).